lib/internal/fs/rimraf.js

```diff
@@ -194,14 +153,11 @@ function rimrafSync(path, options) {
     else
       _unlinkSync(path, options);
   } catch (err) {
-    if (err.code === 'ENOENT')
-      return;
-    if (err.code === 'EPERM')
-      return epermHandlerSync(path, options, err);
-    if (err.code !== 'EISDIR')
-      throw err;
+    if (err.code === 'EISDIR' || err.code === 'EPERM')
+      return _rmdirSync(path, options, err);
```

Comment:
I assume you made this change to match the async implementation (not throwing if the error code is `EISDIR`), I'm a bit surprised it doesn't make any test fail. Do you know if it's because of a lack of coverage? Should we add a test for that?

Reply:
Yes, my intention was to align this with the async implementation. It's still the same behaviour, right? Is there any chance you missed the `!` at line 201 of the previous version?

I do agree that we have a lack of coverage for this code. But I don't really think we can add a test to trigger the `EISDIR` here because:
1. we can get an `EISDIR` only if the `_unlinkSync` call at line 154 is called with `path` pointing to a directory
2. if the `lstatSync` call at line 143 succeeds, then 1 will not happen as `stats?.isDirectory()` at line 151 will equate to `true`
3. 1 will only be triggered if the `lstatSync` call at line 143 failed
4. In case 3 happens, it would mean that `path` is not accessible, so the `_unlinkSync` call at line 154 will throw an `EACCES` (`EPERM` on Windows)

I'm not sure whether we should remove the `EISDIR` check in this PR because I was meaning to make a minimal amount of change to cut off the Windows-specific EPERM code.